### PR TITLE
Add a switch theme option on the dashboard

### LINKED ISSUE #47: [theme] Add a way for the user to switch between a 'light' and a 'dark' theme
As @eric-taix proposed in https://github.com/eric-taix/volley34-mobile/issues/44#issuecomment-649588927, it would be nice to let the user choose between two themes (dark and light).

Where do you think the option should appear ?

### PR DESCRIPTION
**Pull request by thomasFoulon**
Close #47 
I added a switch button to permit the user to switch between a light and a dark theme on the dashboard. There is also a "circle clipper" animation (thanks to a package I found) to have a smooth switching.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -36,6 +37,12 @@ class V34 extends StatefulWidget {
 }
 
 class _V34State extends State<V34> {
+
+  @override
+  void initState() {
```
Comment (thomasFoulon) [inline lib/main.dart:42]: To delete, because useless.

File: lib/pages/dashboard/dashoard_page.dart [hunk 2]
```diff
@@ -1,9 +1,11 @@
+import 'package:animated_theme_switcher/animated_theme_switcher.dart';
 import 'package:collection/collection.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:smooth_page_indicator/smooth_page_indicator.dart';
 import 'package:v34/commons/loading.dart';
 import 'package:v34/commons/router.dart';
+import 'package:v34/main.dart';
```
Comment (thomasFoulon) [inline lib/pages/dashboard/dashoard_page.dart:8]: To remove

**Comment by eric-taix-teads** [comment]
I've played a bit with your branch yesterday, and even if it's a first step we'll have to add a menu somewhere because:
  - switching the theme is not a main action and adding this switch on the main screen adds some noise into the UX. 
  - a second switch must be added to use the auto dark theme feature, if enabled
  - I'd like to display somewhere the contributors of the project (dev + testers)

But this is not required right now: let's merge this one first.

**Comment by thomasFoulon** [comment]
Ok, I can change it now, I'll add a menu !
Do I add a gearwheel icon at the same place as the switch button ?
And what do you think about my light theme ?

**Comment by eric-taix** [comment]
I think that a gearwheel will be fine, but you should also add an app bar
as floating buttons are "reserved" for actions in the current page.
This app bar can be dismissed when the user scrolls down and can be
displayed again when the user scrolls up (no matter the scroll position)

Le lun. 6 juil. 2020 à 13:56, thomasFoulon <notifications@github.com> a
écrit :

**Comment by eric-taix** [comment]
The light theme is a good starting point even if I think it could be more
bright (white)

**Comment by thomasFoulon** [comment]
Here is the new way to switch theme with a menu. Also note that the app bar is dismissed when the user scrolls down and is displayed again when the user scrolls up.

**Comment by eric-taix-teads** [comment]
👏 👏 👏

**Comment by eric-taix-teads** [comment]
When this PR will be merged into master and deployed to the store, can you write a message to beta user group in whatapps with some screenshots to inform them that a new version is available and to get their opinions?

File: lib/pages/dashboard/dashoard_page.dart [hunk 3]
```diff
@@ -61,20 +63,30 @@ class _DashboardPageState extends State<DashboardPage> {
...
+              ),
+              _buildSliverList(state)
+            ],
+          );
+        }
+      )
+    );
+  }
+
+  void _pushPreferences() {
```
Comment (eric-taix-teads) [inline lib/pages/dashboard/dashoard_page.dart:85]: `_gotoPreferencesPage()`? `_push` is really technical

File: lib/pages/dashboard/fav_club_card.dart [hunk 4]
```diff
@@ -153,7 +153,7 @@ class _FavoriteClubCardState extends State<FavoriteClubCard> with AutomaticKeepA
                       textScaleFactor: 1.1,
                       text: TextSpan(
                           text: "X",
-                          style: TextStyle(fontSize: 24.0, fontWeight: FontWeight.bold),
+                          style: TextStyle(fontSize: 24.0, fontWeight: FontWeight.bold, color: Theme.of(context).textTheme.bodyText2.color),
```
Comment (eric-taix-teads) [inline lib/pages/dashboard/fav_club_card.dart:156]: 👍

File: lib/pages/preferences_page.dart [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+          );
+        case 1:
+          return ThemeSwitcher(
+            clipper: ThemeSwitcherCircleClipper(),
+            builder: (context) {
+              return SwitchListTile(
+                contentPadding: EdgeInsets.only(left: 16.0, right: 16.0, bottom: 16.0),
+                title: Text("Mode sombre automatique", style: Theme.of(context).textTheme.bodyText2),
+                subtitle: Text(
+                  "Cette option activera le mode sombre le soir et le désactivera le matin.",
```
Comment (eric-taix-teads) [inline lib/pages/preferences_page.dart:93]:
```suggestion
                  "Cette option active automatiquement le mode sombre le soir et le désactive le matin.",
```

File: lib/theme.dart [hunk 6]
```diff
@@ -71,4 +71,143 @@ class AppTheme {
       ),
     );
   }
+
+  static ThemeData orangeTheme() {
```
Comment (eric-taix-teads) [inline lib/theme.dart:75]: What is this theme for?

File: lib/main.dart (hunk 1 above)
Comment (eric-taix-teads) [reply lib/main.dart:46]: Not more useless

File: lib/main.dart [hunk 7]
```diff
@@ -36,25 +40,74 @@ class V34 extends StatefulWidget {
...
+    super.initState();
+    _preferences = SharedPreferences.getInstance();
+  }
+
+  ThemeData _getTheme(AsyncSnapshot<SharedPreferences> snapshot) {
+    bool automatic = snapshot.data.getBool("automatic_dark_theme") ?? false;
+    bool dark = snapshot.data.getBool("dark_theme") ?? false;
+    DateTime now = DateTime.now();
+    if (automatic) {
+      if (now.hour >= 20 || now.hour < 8) return AppTheme.darkTheme();
```
Comment (eric-taix-teads) [inline lib/main.dart:56]: I have seen this test twice in this PR. You can probably factorize them in a function / method, to be sure that any change will be applied everywhere

**Comment by thomasFoulon** [comment]
Thanks ! :smile: 

I tested the automatic dark theme by changing the time on my phone and it works but only when the application is closed (not in  the cache anymore). Do you know a way to pass in dark theme at 20 o'clock (the hour I chose) even if the application still in the cache ?

And what do you think about 8 o'clock AM and 8 o'clock PM for automatic theme switching ?

File: lib/pages/dashboard/dashoard_page.dart [hunk 8]
```diff
@@ -61,20 +63,30 @@ class _DashboardPageState extends State<DashboardPage> {
...
-              : SizedBox();
-        },
-      ),
+          return CustomScrollView(
+            slivers: <Widget>[
+              SliverAppBar(
+                pinned: false,
+                snap: false,
+                floating: true,
+                title: Text('Tableau de bord'),
```
Comment (eric-taix-teads) [inline lib/pages/dashboard/dashoard_page.dart:72]: As the dashboard is the main page, I think that reusing the application's name is a good pattern. In other pages we will use the a real title

**Comment by eric-taix** [comment]
I think the hours depend on the sun: black theme for the night and light
theme for the day (so it also depends on the month and where you're
leaving, even if it does not really matter for this application)

Le mar. 7 juil. 2020 à 16:19, thomasFoulon <notifications@github.com> a
écrit :

**Comment by thomasFoulon** [comment]
I have taken into consideration all you comments !

**Comment by thomasFoulon** [comment]
Do we implement the sun relative theme switching ?

**Review by Zulu-1540 — CHANGES_REQUESTED** [review]
Spam my WhatsApp

**Review by Zulu-1540 — CHANGES_REQUESTED** [review]
Spam and hacker
Please take it seriously

**Pull request merged**